version/src/lib.rs

```diff
@@ -53,7 +53,9 @@ pub mod error {
         /// Problem with parsing integers
         ParseInt,
         /// Input version unsupported
-        UnsupportedVersion(UnsupportedVersion),
+        UnsupportedVersion(Box<UnsupportedVersion>),
```

Comment:
I'd like to hear what others think about boxing this. This variant has the size of 25 bytes while `ExtraBytesLeft` has the size of 8 bytes. This reduces the size by 17 bytes, but it's not as if there will be a lot of instances of this error

Reply:
Yeah, I've done it just because clippy suggested. Can be removed easily by reverting some commits

Reply:
There should be no ExtraBytesLeft instances returned long-term.

Either leaving the box in, or adding `allow`, is fine. If you leave the box in, make sure to also leave a comment to remove the variant, once partial decoding is fully deprecated.

Reply:
After looking at this `enum` a little, I'm surprised it didn't cause this warning earlier. I think that we crossed a threshold, so the box should stay permanently.